external-crates/move/crates/move-compiler/tests/move_2024/typing/lambda_subtyping_usage_respects_annotations.exp

```diff
@@ -6,9 +6,7 @@ error[E04006]: invalid subtype
    │          │
    │          Invalid mutation. Expected a mutable reference
    │          Expected: '&mut _'
-   ·
```

Comment:
Not sure what have caused this and could use some insights hare

Reply:
This is because the lambda block now has a name. Named blocks need their types determined by the typing context due to non-local control flow, which is lossy: we can't know if the specific type came from a specific statement or multiple, so we just report the whole block. In this case, the block's location is now `$f`. To fix this, we need a dual to `ExpandedMacro` called `ExpandedLambda`, or better, add some `IdeAnnotation(IdeInfo, Box<Exp>)` to the grammar that we can extend later.

Reply:
After swapping elaboration order, the problem disappeared actually